system/lib/libc/musl/src/unistd/dup2.c

```diff
@@ -14,7 +14,7 @@ int dup2(int old, int new)
 #else
 	if (old==new) {
 #ifdef __EMSCRIPTEN__
-		r = __wasi_fd_is_valid(old) ? 0 : -1;
+		r = __wasi_fd_is_valid(old) ? 0 : -EBADF;
```

Comment:
Is this a separate bugfix from this PR?

Reply:
Yes indeed, let me see if I can split that out.

Reply:
Actually this is bugfix, but it doesn't make sense to land separately really since this is basically in dead code.   Prior to this PR `SYS_dup2` is defined so this entire block of code is ifdef'd out.   By removing `SYS_dup2` in this PR we are forced to fix this existing bug. 

I guess I could split out the removal of dup2.... that could work.